app/models/storage_location.rb

```diff
@@ -60,6 +60,16 @@ def size
     inventory_items.sum(:quantity)
   end
 
+  def item_total_value(item_id)
+    value_in_cents = Item.find(item_id).try(:value_in_cents)
+    quantity = item_total(item_id)
+    (value_in_cents / 100) * quantity
+  end
+
+  def inventory_total_value
+    inventory_items.map { |item| item_total_value(item.item_id) }.reduce(:+)
```

Comment:
This is going to have an N+1 query problem. It'll pull all the `inventory_items` and then individually do a separate query on each one on line 64. 

Since you're only using `item_total_value` here, you can combine these two methods together, like this:

```ruby
inventory_items.map do |inventory_item| 
  value_in_cents = Item.find(inventory_item.item_id).try(:value_in_cents)
  quantity = item_total(inventory_item.item_id)
  (value_in_cents / 100) * quantity
end.reduce(:+)
```
And then join the item association to it:

```ruby
# Add a `joins` to the front so we don't get an N+1 issue
inventory_items.joins(:item).map do |inventory_item| 
  value_in_cents = inventory_item.item.try(:value_in_cents)
  # Since this is an instance method, we're only going to get the quantity for this storage location; we don't need item_total
  # quantity = item_total(inventory_item.item_id)
  # We have the quantity right in `inventory_item.quantity`
  # on the off-chance that the value is 0 or nil, this would be a div/0 exception
  # (value_in_cents / 100) * quantity
  # Instead, let's just keep it in cents and divide later
  value_in_cents * inventory_item.quantity
end.reduce(:+)
```
The value you get will still be in cents, but you can now check if it's greater than 0 and then divide by 100 (or leave it in cents and just change the method to something like `inventory_total_value_in_cents` or something

Reply:
Awesome! Thanks for the tip!